docs: refactor style guide example 03-06

danielsogl opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Style guide example 03-06 uses the old Angular Http module


## What is the new behavior?
Updated the example to use thew new HttpClient 


## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

danielsogl commented:
cc @brandonroberts

vicb commented:
Failing the CI with the error:

```
ERROR in src/03-06/app/heroes/shared/hero.service.avoid.ts(24,32): error TS2339: Property 'data' does not exist on type 'Hero'.
src/03-06/app/heroes/shared/hero.service.avoid.ts(29,32): error TS2339: Property 'data' does not exist on type 'Hero[]'.
```

danielsogl commented:
@vicb should be fixed now

gkalpak commented on aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts:
```diff
@@ -16,18 +16,18 @@ export class HeroService {
-    private http: Http
-  ) { }
+    private http: HttpClient
+  ) {}
```
Keep the space (for consistency).

gkalpak commented on aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts:
```diff
@@ -16,18 +16,18 @@ export class HeroService {
   }
-
 }
-
```
Other than the imports, the rest of this file should be identical to [hero.service.ts][1].

[1]: https://github.com/danielsogl/angular/blob/2047abe91413aa13f10b549efa989a6852b81ecd/aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.ts

gkalpak commented on aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.ts:
```diff
@@ -16,17 +17,17 @@ export class HeroService {
-    return this.http.get(`api/heroes`).pipe(
-      map(response => response.json() as Hero[]));
+    return this.http.get<Hero[]>(`api/heroes`).pipe(
+      map(response => response as Hero[]));
```
The `pipe(map(...))` is no longer necessary (here and above).

danielsogl commented:
@gkalpak Fixed

gkalpak commented on aio/content/examples/styleguide/src/03-06/app/heroes/shared/hero.service.avoid.ts:
```diff
@@ -16,18 +15,14 @@ export class HeroService {
-      map(response => response.json().data as Hero[]));
+    return this.http.get<Hero[]>(`api/heroes`);
   }
-
```
This line should stay 😇

danielsogl replied:
When I open the file there is a free line after the last curly bracket and in GitHub Web too.

danielsogl replied:
Ahhh damit 😅

gkalpak reviewed (approved):
One last nit. LGTM otherwise (assuming CI is green) 👍

danielsogl committed:
docs: refactor style guide example 03-06


docs: refactor style guide example 03-06


docs: refactor style guide example 03-06


docs: refactor style guide example 03-06

vicb commented:
Thanks for the update, merging again.
